### DISCUSSION TITLE
useRouteMatch,connect is not working while upgrading umijs v4

### DISCUSSION THREAD

**Discussion by Khadhiri786** [umijs/umi#11368] [General]
`import {useRouteMatch,connect,request} from 'umi'` is not working in umijs v4

**Comment by fz6m** [comment]
1. `useRouteMatch` : Umi 4 using [react-router v6](https://reactrouter.com/) , you should use:

    ```ts
    import { useParams } from 'umi'
    
    export default function Page() {
      // https://reactrouter.com/en/main/hooks/use-params
      // you can import `react-router` methods from 'umi'
      const { userId } = useParams();
    
      // ...
    }
    ```

2. `connect` / `request` : In umi4, the umi 3 plugins is not included by default, you should install and enable it

    ```bash
      pnpm i -D @umijs/plugins
    ```
    
    ```ts
    // .umirc.ts
    
    export default {
      plugins: [
        '@umijs/plugins/dist/dva',
        '@umijs/plugins/dist/request'
      ],
      dva: {},
      request: {}
    }
    ```

**Comment by Khadhiri786** [comment]
We are not using dot file for umi till date. Can we add the above code snippet in the config.ts? where we have done configurations for umi?

**Reply by fz6m** [reply]
using either `.umirc.ts` or `config/config.ts` is the same.
